### DISCUSSION TITLE
blog/zh/gpt-embeddings/

### DISCUSSION THREAD

**Discussion by giscus** [guangzhengli/blog-comments#13] [Announcements]
# blog/zh/gpt-embeddings/

Ladder@在过去几个月的时间中，我们似乎正处于人工智能的革命中。除了大多数人了解的 OpenAI ChatGPT 之外，许多非常新颖、有趣、实用的 AI 应用也是层出不穷，并且在使用这些应用时时，笔者也确确实实的感受到了生产力的提高。
但是关于 GPT 应用的开发知识和路线，目前似乎还没有太多的资料，所以笔者决定将自己的一些经验和思考整理成一个系列，希望能够帮助到大家。
本篇文章主要介绍的是 GPT Embeddings 相关应用的开发，在今年 4 月份的时候，笔者因为开发 ChatFiles 这个开源项目，从而学习了 GPT Embeddings 相关的技术知识，但是由于笔者的时间精力有限，所以一直没有机会将这些知识整理成一篇文章，直到最近笔者又因为有了新的想法，开源了 VectorHub 这个同样基于 GPT Embeddings 技术的项目，进而对 GPT Embeddings 的技术知识有了更深入的了解，所以就有了这一篇分享。

https://guangzhengli.com/blog/zh/gpt-embeddings/

**Comment by helloChenLei** [comment]
🤔 文章是还没有写完吗？
我看最后一句是「上一篇中，我们主要介绍了向量数据库，而」

**Reply by guangzhengli** [reply]
sorry 是的，还没写完，不小心推上来了哈哈哈哈...

**Comment by hw95031** [comment]
赶紧更新呀

**Comment by 17336** [comment]
写的真好，每期必看！

**Comment by ycjcl868** [comment]
大佬求加个联系方式

**Reply by ycjcl868** [reply]
kylinJCL

**Comment by joepeak** [comment]
大佬，很中意你的这篇文章，想转发到我的 blog上面，会注明转自你的网站，也会提供连接跳转到你这里，可以吗？

**Reply by guangzhengli** [reply]
可以的，转载标明出处就可以滴，谢谢！

**Reply by joepeak** [reply]
好的，会注明你的链接的，谢谢

**Comment by hw95031** [comment]
同感，大佬的文章质量挺高，两篇关于chatgpt的我都看了



---Original---
From: ***@***.***&gt;
Date: Thu, Sep 14, 2023 14:58 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [guangzhengli/blog-comments] blog/zh/gpt-embeddings/ (Discussion#13)




 
写的真好，每期必看！

**Comment by d2clab** [comment]
大佬好。我是不会写代码的麻瓜一枚，想问问：我虽然不会写能跑起来的代码，但是把函数定义套用在 prompt 里面，也能实现很好的结构化输出——这么说可以吗？大佬有见过什么这种用例吗？

**Comment by gaoxiaobo0513** [comment]
干货满满，文章质量真的非常高，且实用

**Comment by hydah** [comment]
博主语言功底深厚！！

**Comment by StLeoX** [comment]
产品文档有吗？
stleox@qq.com

**Comment by noneback** [comment]
"这个过程就是 GPT Embeddings 应用开发核心，它的核心思想是通过向量检索的方式检索与问题最相关的文本段，从而绕过 GPT tokens 的限制。"  这句话该怎么理解呢？怎么绕过token的机制的呢？通过向量检索的方式，本质上是一种预处理，意思是通过这种预处理来过滤不关联的token？

**Reply by guangzhengli** [reply]
好问题，确实是一种预处理方式，绕过这个词表达的确实不太准确。

更准确的是在 token 的限制下，尽量过滤出最关联的文本段，例如 1w 字的文章中根据提出问题的关联性，过滤出几个连续的段落，总共 1k 字，这样的文本段转成 token 后，不会触发各大厂商对于 token 的限制。又能达到过滤出的文本，能够回答用户的问题这样的效果。

**Comment by 9527dong** [comment]
写的真不错，看完很有收获

**Comment by adiligentpiggy** [comment]
大佬牛牛牛！！！

**Comment by luyifo** [comment]
真的收获满满

**Comment by KingCoding518** [comment]
登录只为评论，大佬功力深厚，手动赞一个

**Comment by zhanghuanzhong** [comment]
很棒，看到了很多博主的思考，干货！

**Comment by zhanghuanzhong** [comment]
很棒，看到了很多博主的思考，干货！
